server/src/main/java/dev/slimevr/config/CurrentVRConfigConverter.java

```diff
@@ -18,188 +19,212 @@ public ObjectNode convert(
-					offsetsNode.set("handDistanceZ", offsetsNode.get("controllerDistanceZ"));
+			if (version < 3) {
+				// Check for out-of-bound filtering amount
+				ObjectNode filtersNode = (ObjectNode) modelData.get("filters");
+				if (filtersNode != null && filtersNode.get("amount").floatValue() > 2f) {
+					filtersNode.set("amount", new FloatNode(0.2f));
 				}
 			}
-		}
-		if (version < 7) {
-			// Chest, hip, and elbow offsets now go the opposite direction
-			ObjectNode skeletonNode = (ObjectNode) modelData.get("skeleton");
-			if (skeletonNode != null) {
-				ObjectNode offsetsNode = (ObjectNode) skeletonNode.get("offsets");
-				if (offsetsNode != null) {
-					if (offsetsNode.get("chestOffset") != null) {
+			if (version < 4) {
+				// Change mountingRotation to mountingOrientation
+				ObjectNode oldTrackersNode = (ObjectNode) modelData.get("trackers");
+				if (oldTrackersNode != null) {
+					var trackersIter = oldTrackersNode.iterator();
+					var fieldNamesIter = oldTrackersNode.fieldNames();
+					ObjectNode trackersNode = nodeFactory.objectNode();
+					String fieldName;
+					while (trackersIter.hasNext()) {
+						ObjectNode node = (ObjectNode) trackersIter.next();
+						fieldName = fieldNamesIter.next();
+						node.set("mountingOrientation", node.get("mountingRotation"));
+						node.remove("mountingRotation");
+						trackersNode.set(fieldName, node);
+					}
+					modelData.set("trackers", trackersNode);
+				}
+			}
+			if (version < 5) {
+				// Migrate old skeleton offsets to new ones
+				ObjectNode skeletonNode = (ObjectNode) modelData.get("skeleton");
+				if (skeletonNode != null) {
+					ObjectNode offsetsNode = (ObjectNode) skeletonNode.get("offsets");
+					if (offsetsNode != null) {
+						// torsoLength, chestDistance and waistDistance become
+						// chestLength, waistLength and hipLength.
+						float torsoLength = 0, chestDistance = 0, waistDistance = 0;
+						JsonNode torsoNode = offsetsNode.get("torsoLength");
+						if (torsoNode != null)
+							torsoLength = torsoNode.floatValue();
+						JsonNode chestNode = offsetsNode.get("chestDistance");
+						if (chestNode != null)
+							chestDistance = chestNode.floatValue();
+						JsonNode waistNode = offsetsNode.get("waistDistance");
+						if (waistNode != null)
+							waistDistance = waistNode.floatValue();
+						offsetsNode.set("chestLength", offsetsNode.get("chestDistance"));
 						offsetsNode
 							.set(
-								"chestOffset",
-								new FloatNode(-offsetsNode.get("chestOffset").floatValue())
+								"waistLength",
+								new FloatNode(torsoLength - chestDistance - waistDistance)
 							);
+						offsetsNode.set("hipLength", offsetsNode.get("waistDistance"));
+						offsetsNode.remove("torsoLength");
+						offsetsNode.remove("chestDistance");
+						offsetsNode.remove("waistDistance");
+
+						// legsLength and kneeHeight become
+						// upperLegLength and lowerLegLength
+						float legsLength = 0, kneeHeight = 0;
```

Comment:
Same as the comment before about using the default length values, is that possible?